### PR TITLE
Move some BoundingBox access functions to the header.

### PR DESCRIPTION
**Pull request by drwells**
It's useful to be able to inline these.

### CODE REVIEW

File: include/deal.II/base/bounding_box.h [hunk 1]
```diff
@@ -437,6 +437,24 @@ inline BoundingBox<spacedim, Number>::BoundingBox(const Container &points)
 
 
 
+template <int spacedim, typename Number>
+inline std::pair<Point<spacedim, Number>, Point<spacedim, Number>> &
+BoundingBox<spacedim, Number>::get_boundary_points()
```
Comment (bangerth) [inline include/deal.II/base/bounding_box.h:442]: Unrelated question, but why would we give write access to this information?

**Review by bangerth — APPROVED** [review]

File: include/deal.II/base/bounding_box.h (hunk 1 above)
Comment (luca-heltai) [reply include/deal.II/base/bounding_box.h:442]: It is required by boost to have write access to bonding boxes to be able to convert from and to deal.II bounding boxes in its algorithms.

Comment (bangerth) [reply include/deal.II/base/bounding_box.h:442]: I see. That's a nuisance...

Is there no way to just create a new BB object and then use the assignment operator? Not a question for this PR, but if you wanted to think about it (or if you wanted to point me to the place where this is needed).

**Pull request merged**